src/Agoda.IoC.NetCore/NetCoreKeyedComponentResolver.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Agoda.IoC.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Agoda.IoC.NetCore
+{
+    public class NetCoreKeyedComponentResolver<T> : IKeyedComponentResolver<T>
+    {
+        private readonly IServiceProvider _container;
+        private readonly IDictionary<string, Type> _registrations;
+
+        public NetCoreKeyedComponentResolver(IServiceProvider container, NetCoreKeyedRegistrations<T> registrations)
+        {
+            _container = container;
+            _registrations = registrations.Registrations;
+        }
+
+        public T Resolve(string key)
+        {
+            if (!_registrations.TryGetValue(key, out var implementationType))
+                throw new ArgumentException($"Service name '{key}' is not registered");
+            return (T)_container.GetService(implementationType);
+        }
+
+        public bool IsRegistered(string key)
+        {
+            return !_registrations.TryGetValue(key, out var implementationType);
+        }
```

Comment:
```public bool IsRegistered(string key) => _registrations.TryGetValue(key, out var implementationType);```

Reply:
there's a `!` that shouldn't be there

Reply:
nice catch, htf did that pass the unit test is what i need to check to